### PR TITLE
Authentication GUI MVP

### PR DESCRIPTION
**Pull request by Wuzado**
This is a basic implementation of account authentication in the Tauri GUI, complete with a basic interface made for testing, using the Theseus library as the backend/store.

Creating this as a draft PR to discuss how to properly merge the integration with the Theseus library.

### CODE REVIEW

File: theseus_gui/src/routes/settings/accounts.svelte [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      color="primary"
+      class="button-row__items"
+      disabled={user === undefined || user.uninitialized}
+      on:click={userRemove}>Log out</Button
+    >
+  </div>
+</div>
+
+<style lang="postcss">
+  .settings-accounts {
```
Comment (pauliesnug) [inline theseus_gui/src/routes/settings/accounts.svelte:71]: are inline styles for this necessary?

Comment (Wuzado) [reply theseus_gui/src/routes/settings/accounts.svelte:71]: It's just a demo/debugging interface, loosely based on existing pages and Omorphia. Can this be improved? Yes, but this wasn't the original purpose behind this PR.

File: theseus_gui/src/routes/settings/accounts.svelte [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<script>
+  import { Button } from 'omorphia';
+  import { invoke } from '@tauri-apps/api/tauri';
+  import { onMount } from 'svelte';
+
+  let user = { uninitialized: true };
+
+  function fetchUser() {
```
[the remark this answers is not in the document]
Comment (Wuzado) [reply theseus_gui/src/routes/settings/accounts.svelte:8]: To be frank, I think the entire fetching on the frontend may need to be rewritten. I'm not exactly too experienced with Svelte, so I had to fight a lot to get things to get things to work in a basic state. Type safety is always nice to have.

File: theseus_gui/src-tauri/src/main.rs [hunk 3]
```diff
@@ -12,7 +14,11 @@ fn main() {
     let ctx = tauri::generate_context!(); // Run `pnpm build:web` (builds the web app) to get rid of the error.
 
     tauri::Builder::default()
-        .invoke_handler(tauri::generate_handler![])
+        .invoke_handler(tauri::generate_handler![
```
[the remark this answers is not in the document]
Comment (Wuzado) [reply theseus_gui/src-tauri/src/main.rs:17]: Looks interesting.

**Comment by Geometrically** [comment]
Closing this as this will be implemented in a separate PR and we are using Vue for the UI now

**Pull request closed without merging**